src/analysis/retail/evoker/preservation/modules/talents/TimeOfNeed.tsx

```diff
@@ -0,0 +1,234 @@
+      }
+    });
+
+    this.spawns.push(currentEvent);
+  }
+
+  parseDamageTaken(tonSpawn: tonEvent, eventList: WCLEventsResponse) {
+    //Check that ToN actually did any healing or if it bugged out
+    if (tonSpawn.verdantEmbrace === null && tonSpawn.livingFlames.length === 0) {
+      return Result.Bug;
+    }
+    //If there is no damage events, then Time of Need didn't save the player from anything
+    if (eventList.events.length === 0) {
+      return Result.Not;
+    }
+    const hits = eventList.events as DamageEvent[];
+    //Iterate all the damage taken events during the time window
+    for (const hit of hits) {
+      //Check that the info from the damage event is valid
+      if (hit.hitPoints !== undefined && hit.targetID === tonSpawn.target) {
+        //If the hitpoints after the hit are equal to 0, the player died anyway
+        if (hit.hitPoints === 0) {
+          return Result.Dead;
+        }
+        //Initialize the healing from Time of Need with the VE
+        let tonHealingAtThisPoint = tonSpawn.verdantEmbrace?.amount || 0;
+        //If there is no living flames casted, or this hit was before the first living flame landed
+        if (
+          tonSpawn.livingFlames.length === 0 ||
+          hit.timestamp < tonSpawn.livingFlames[0].timestamp
+        ) {
+          //Check if the player would've died from this hit without the Verdant Embrace healing
+          if (hit.hitPoints - tonHealingAtThisPoint <= 0) {
+            return Result.Proc; //Person was saved by the VE
+          }
+        } else {
+          //Add up all the healing done by the living flames that happened before this hit
+          tonSpawn.livingFlames.forEach((livingFlame) => {
+            if (livingFlame.timestamp < hit.timestamp) {
+              tonHealingAtThisPoint += livingFlame.amount;
+            }
+          });
+          //If the hit points after this hit minus the total ToN healing done to this point is equal or less than 0, then he was saved
+          if (hit.hitPoints > 0 && hit.hitPoints - tonHealingAtThisPoint <= 0) {
+            return Result.Long; //Mark as long save
+          }
+        }
+      }
+    }
+    return Result.Not;
+  }
+
+  async load() {
+    //Run every ToN spawn through parseDamageTaken()
+    for (const spawn of this.spawns) {
+      const damageTakenEvents = await fetchWcl<WCLEventsResponse>(
+        `report/events/damage-taken/${this.owner.report.code}`,
+        {
+          start: spawn.summon.timestamp,
+          end: spawn.summon.timestamp + 8000,
+        },
+      );
+      spawn.result = this.parseDamageTaken(spawn, damageTakenEvents);
+    }
+  }
+
+  statistic() {
+    return (
+      <LazyLoadStatisticBox
+        category={STATISTIC_CATEGORY.TALENTS}
+        position={STATISTIC_ORDER.OPTIONAL(60)}
+        loader={this.load.bind(this)}
+        label={
+          <Trans>
+            <SpellLink spell={TALENTS_EVOKER.TIME_OF_NEED_TALENT} /> events
+          </Trans>
```

Comment:
I think we stopped using these for individuals modules, should be fine to remove
```suggestion
            <SpellLink spell={TALENTS_EVOKER.TIME_OF_NEED_TALENT} /> events
```